src/Validation/Validator.php

```diff
@@ -295,8 +295,24 @@ public function hasField(string $name): bool
      * @param string $name The name under which the provider should be set.
      * @param object|string $object Provider object or class name.
      * @return $this
+     * @deprecated In favour of addProvider()
```

Comment:
```suggestion
     * @deprecated 4.2.0 Use `addProvider()` instead
```

Reply:
Any reason we wouldn't add a deprecation warning here? We could add a rector rule to make upgrading easier.

Reply:
No reason from my side.  I thought this might ease people into the change in a less "holy cow my code is throwing weird notices at me after upgrading" type of way.  But if you're happy with having a deprecation warning (i.e. if it's standard practice in a situation like this), I'll add it.